atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,690 @@
+
+            if workflow_properties_dict.get(property_type_key):
+                workflow_properties_dict[property_type_key].append(property_dict)
+            else:
+                workflow_properties_dict[property_type_key] = [property_dict]
+
+        return workflow_properties_dict
+
+    def parse_workflow(self):
+
+        workflow_parameters = self._parameter_info.get('workflow').get('molecular_dynamics')
+        # TODO should store parameters that do not match the enum vals as x_h5MD params,
+        # not sure how with MDParser??
+        if workflow_parameters is None:
+            return
+
+        workflow_results = {}
+        ensemble_average_observables = self._observable_info.get('ensemble_average')
+        ensemble_property_dict = {
+            'property_type_key': 'ensemble_properties',
+            'property_type_value_key': 'ensemble_property_values',
+            'properties_known': {'radial_distribution_functions': 'radial_distribution_function_values'},
+            'property_keys_list': EnsembleProperty.m_def.all_quantities.keys(),
+            'property_value_keys_list': EnsemblePropertyValues.m_def.all_quantities.keys()
+        }
+        workflow_results.update(
+            self.get_workflow_properties_dict(ensemble_average_observables, **ensemble_property_dict)
+            )
+        correlation_function_observables = self._observable_info.get('correlation_function')
+        correlation_function_dict = {
+            'property_type_key': 'correlation_functions',
+            'property_type_value_key': 'correlation_function_values',
+            'properties_known': {'mean_squared_displacements': 'mean_squared_displacement_values'},
+            'property_keys_list': CorrelationFunction.m_def.all_quantities.keys(),
+            'property_value_keys_list': CorrelationFunctionValues.m_def.all_quantities.keys()
+        }
+        workflow_results.update(
+            self.get_workflow_properties_dict(correlation_function_observables, **correlation_function_dict)
+            )
+        self.parse_md_workflow(dict(method=workflow_parameters, results=workflow_results))
+
+    def init_parser(self):
+
+        self._data_parser.mainfile = self.filepath
+
+    def parse(self, filepath, archive: EntryArchive, logger):
+
+        self.filepath = os.path.abspath(filepath)
+        self.archive = archive
+        self.logger = logging.getLogger(__name__) if logger is None else logger
+        self._maindir = os.path.dirname(self.filepath)
+        self._h5md_files = os.listdir(self._maindir)
+        self._basename = os.path.basename(filepath).rsplit('.', 1)[0]
+
+        self.init_parser()
+
+        if self._data_parser.filehdf5 is None:
+            self.logger.warning('hdf5 file missing in H5MD Parser.')
+            return
+
+        # Define some useful paths within the hdf5 file
+        self._path_group_particles_all = 'particles.all'
```

Comment:
are these not supposed to be in __init__

Reply:
I moved these up where the other variables are specified within parse(), not sure if this is what you meant.

Reply:
do these change? if not put them in 'def __init__' like all the other constants.

Reply:
oh :facepalm: got it, done.